src/api/python/cvc4.pxi

```diff
@@ -1503,6 +1504,98 @@ cdef class Term:
         term.cterm = self.cterm.iteTerm(then_t.cterm, else_t.cterm)
         return term
 
+    def toPythonObj(self):
+        '''
+        Converts a constant value Term to a Python object.
+        Requires isConst to hold.
+
+        Currently supports:
+          Boolean -- returns a Python bool
+          Int     -- returns a Python int
+          Real    -- returns a Python Fraction
+          BV      -- returns a Python int (treats BV as unsigned)
+          Array   -- returns a Python dict
+                  -- the constant array base is the value for the string '*'
+        '''
+
+        if not self.isConst():
+            raise RuntimeError("Cannot call toPythonObj on a non-const Term")
+
+        string_repr = self.cterm.toString().decode()
+        assert string_repr
+        sort = self.getSort()
+        res = None
+        if sort.isBoolean():
+            if string_repr == "true":
+                res = True
+            else:
+                assert string_repr == "false"
+                res = False
+        elif sort.isInteger():
+            is_neg = False
+            updated_string_repr = string_repr
+            if string_repr[:1] == '(-':
+                is_neg = True
+                updated_string_repr = string_repr[2:-1]
+            try:
+                res = int(updated_string_repr)
+            except:
+                raise ValueError("Failed to convert {} to an int".format(string_repr))
+
+            if is_neg:
+                res = -res
+        elif sort.isReal():
+            is_neg = False
+            updated_string_repr = string_repr
+            if string_repr[:1] == '(-':
+                is_neg = True
+                updated_string_repr = string_repr[2:-1]
+
+            try:
+                # expecting format (/ a b)
+                splits = updated_string_repr.split()
+                assert len(splits) == 3
+                num = int(splits[1])
+                den = splits[2]
+                if den[-1] == ')':
+                    den = den[:-1]
+                den = int(den)
+                res = Fraction(num, den)
+            except:
+                raise ValueError("Failed to convert {} to a Fraction".format(string_repr))
+
+            if is_neg:
+                res = -res
+        elif sort.isBitVector():
+            # expecting format #b<bits>
+            assert string_repr[:2] == "#b"
+            python_bin_repr = "0" + string_repr[1:]
+            try:
+                res = int(python_bin_repr, 2)
+            except:
+                raise ValueError("Failed to convert bitvector {} to an int".format(string_repr))
+        elif sort.isArray():
+            res = {}
```

Comment:
You could maybe use a defaultdict here so that the dict returns by default the base case instead of the `*` entry.

Reply:
That's very clever, I like it! Added in 3ba57ed7b00f65e1c1263934a07a90f62e67e645.